mlflow/tracking/client.py

```diff
@@ -648,6 +649,29 @@ def end_trace(
 
         self.end_span(request_id, root_span_id, outputs, attributes, status)
 
+    def _upload_trace_spans_as_tag(self, trace_info: TraceInfo, trace_data: TraceData):
+        # When a trace is logged, we set a mlflow.traceSpans tag via SetTraceTag API
+        # https://databricks.atlassian.net/browse/ML-40306
+        parsed_spans = []
+        for span in trace_data.spans:
+            parsed_span = {}
+
+            parsed_span["name"] = span.name
+            parsed_span["type"] = span.get_attribute(SpanAttributeKey.SPAN_TYPE)
+            span_inputs = span.get_attribute(SpanAttributeKey.INPUTS)
+            if span_inputs and isinstance(span_inputs, dict):
+                parsed_span["inputs"] = list(span_inputs.keys())
```

Comment:
What if span_inputs is not a dict, do we want to store it or not?

Reply:
Same for outputs

Reply:
If span inputs or outputs is not a dict, we don't want to save it